msat/src/msat_solver.cpp

```diff
@@ -846,7 +847,22 @@ Term MsatSolver::substitute(const Term term,
 
 void MsatSolver::dump_smt2(FILE * file) const
 {
-  throw NotImplementedException("Can't dump all assertions to a file yet");
+  size_t num_asserted;
+  msat_term * assertions = msat_get_asserted_formulas(env, &num_asserted);
+  msat_term * ait = assertions;
```

Comment:
why do we need this extra pointer

Reply:
Oh yeah, good catch. That was left over from something else I was trying. Fixed with 75bb8e4bc5d592bc446f0f4debfa68caf44ada46